### DISCUSSION TITLE
aimp discord presence

### DISCUSSION THREAD

**Discussion by prototype464** [Exle/aimp-discord-presence#36] [Similar developments]
I tried MusicBee at some point and its Discord Rich Presence plugin is a lot better, showing album covers and the current timestamp, among other things. This plugin is missing those, any chance someone smarter than me could fork it and make those updates? That would be awesome.

Thank you!

**Comment by FatalFit** [comment]
Tried the plugin today then saw this, what a coincidence, it would be amazing if someone tried to bring this back to life

**Comment by Chris3461** [comment]
I hope someone can show the album cover :(

**Comment by iBowie** [comment]
> I hope someone can show the album cover :(

![image](https://github.com/Exle/AIMP-Discord-Presence/assets/31174978/d26eeea0-04bd-486c-9f6c-8d893b0e4f82)

I may have some advances in that regard!

Note: this is a dynamic link, not discord asset. So it's not limited by this.

**Comment by kwotaq** [comment]
> > I hope someone can show the album cover :(
> 
> ![image](https://private-user-images.githubusercontent.com/31174978/301672168-d26eeea0-04bd-486c-9f6c-8d893b0e4f82.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY5NzE5NTgsIm5iZiI6MTcwNjk3MTY1OCwicGF0aCI6Ii8zMTE3NDk3OC8zMDE2NzIxNjgtZDI2ZWVlYTAtMDRiZC00ODZjLTlmNmMtOGQ4OTNiMGU0ZjgyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjAzVDE0NDczOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY1NWRjNTU5ZDkwMThkNjZmYWZhZjYxYmFhODcwOWYxY2EyYzU5M2Y3NmJmZjg3ZDg1OWJlNzZhNWZlZDYwODEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.IuKiQWyXc1f5lAU49-EpyPGrEJL8qSz-ZSnyBB0hcU8)
> 
> I may have some advances in that regard!
> 
> Note: this is a dynamic link, not discord asset. So it's not limited by this.

Please keep us updated! Would love to see this feature implemented

**Comment by iBowie** [comment]
Currently my main roadblock is that I can't find a cover art service which URL will not be longer than 128 characters.

I tried MusicBrainz with CovertArchive, but it requires THREE separate web requests and resulting link is too long for Discord to handle. Maybe someone could lend me a hand with this and help me find a service like this?

For the proof-of-concept I used my own VPS, just to make sure it will work.

**Comment by kwotaq** [comment]
> Currently my main roadblock is that I can't find a cover art service which URL will not be longer than 128 characters.
> 
> I tried MusicBrainz with CovertArchive, but it requires THREE separate web requests and resulting link is too long for Discord to handle. Maybe someone could lend me a hand with this and help me find a service like this?
> 
> For the proof-of-concept I used my own VPS, just to make sure it will work.

Does the largeimagekey accept any link? If the character size itself is the issue couldn't you use a link shortener? That would be even more web requests I guess but it may solve the problem. Does it only accept links? Could you perhaps index the queries into a local database and parse it from there? Not really familiar with the limitations of the Discord rich presence stuff but I'm just throwing out ideas to see if I could help in some way, sorry if my suggestions are uninformed.

**Comment by iBowie** [comment]
> > Currently my main roadblock is that I can't find a cover art service which URL will not be longer than 128 characters.
> > I tried MusicBrainz with CovertArchive, but it requires THREE separate web requests and resulting link is too long for Discord to handle. Maybe someone could lend me a hand with this and help me find a service like this?
> > For the proof-of-concept I used my own VPS, just to make sure it will work.
> 
> Does the largeimagekey accept any link? If the character size itself is the issue couldn't you use a link shortener? That would be even more web requests I guess but it may solve the problem. Does it only accept links? Could you perhaps index the queries into a local database and parse it from there? Not really familiar with the limitations of the Discord rich presence stuff but I'm just throwing out ideas to see if I could help in some way, sorry if my suggestions are uninformed.

Unfortunately, Discord requires direct image link, otherwise it wouldn't work.

**Comment by kwotaq** [comment]
> > > Currently my main roadblock is that I can't find a cover art service which URL will not be longer than 128 characters.
> > > I tried MusicBrainz with CovertArchive, but it requires THREE separate web requests and resulting link is too long for Discord to handle. Maybe someone could lend me a hand with this and help me find a service like this?
> > > For the proof-of-concept I used my own VPS, just to make sure it will work.
> > 
> > 
> > Does the largeimagekey accept any link? If the character size itself is the issue couldn't you use a link shortener? That would be even more web requests I guess but it may solve the problem. Does it only accept links? Could you perhaps index the queries into a local database and parse it from there? Not really familiar with the limitations of the Discord rich presence stuff but I'm just throwing out ideas to see if I could help in some way, sorry if my suggestions are uninformed.
> 
> Unfortunately, Discord requires direct image link, otherwise it wouldn't work.

Have you tried Spotify or Youtube Music as a source for album covers? Also personally I have all my music downloaded with the covers in its metadata, is your implementation parsing local files or only searching the web for them?

**Comment by iBowie** [comment]
> > > > Currently my main roadblock is that I can't find a cover art service which URL will not be longer than 128 characters.
> > > > I tried MusicBrainz with CovertArchive, but it requires THREE separate web requests and resulting link is too long for Discord to handle. Maybe someone could lend me a hand with this and help me find a service like this?
> > > > For the proof-of-concept I used my own VPS, just to make sure it will work.
> > > 
> > > 
> > > Does the largeimagekey accept any link? If the character size itself is the issue couldn't you use a link shortener? That would be even more web requests I guess but it may solve the problem. Does it only accept links? Could you perhaps index the queries into a local database and parse it from there? Not really familiar with the limitations of the Discord rich presence stuff but I'm just throwing out ideas to see if I could help in some way, sorry if my suggestions are uninformed.
> > 
> > 
> > Unfortunately, Discord requires direct image link, otherwise it wouldn't work.
> 
> Have you tried Spotify or Youtube Music as a source for album covers? Also personally I have all my music downloaded with the covers in its metadata, is your implementation parsing local files or only searching the web for them?

Spotify has closed API, so does YouTube, and their links are most certainly longer than 128 characters (due to CDN and a lot of parameters and long names).
You can't just send image to Discord as cover, since you can only use either pre-loaded Asset on Discord Developer Portal, or direct image link that is no longer than 128 characters. So yes, I used the web to search for images. However, I am still in search of a music service that would work well with Discord's limitations.

**Comment by kwotaq** [comment]
> > > > > Currently my main roadblock is that I can't find a cover art service which URL will not be longer than 128 characters.
> > > > > I tried MusicBrainz with CovertArchive, but it requires THREE separate web requests and resulting link is too long for Discord to handle. Maybe someone could lend me a hand with this and help me find a service like this?
> > > > > For the proof-of-concept I used my own VPS, just to make sure it will work.
> > > > 
> > > > 
> > > > Does the largeimagekey accept any link? If the character size itself is the issue couldn't you use a link shortener? That would be even more web requests I guess but it may solve the problem. Does it only accept links? Could you perhaps index the queries into a local database and parse it from there? Not really familiar with the limitations of the Discord rich presence stuff but I'm just throwing out ideas to see if I could help in some way, sorry if my suggestions are uninformed.
> > > 
> > > 
> > > Unfortunately, Discord requires direct image link, otherwise it wouldn't work.
> > 
> > 
> > Have you tried Spotify or Youtube Music as a source for album covers? Also personally I have all my music downloaded with the covers in its metadata, is your implementation parsing local files or only searching the web for them?
> 
> Spotify has closed API, so does YouTube, and their links are most certainly longer than 128 characters (due to CDN and a lot of parameters and long names). You can't just send image to Discord as cover, since you can only use either pre-loaded Asset on Discord Developer Portal, or direct image link that is no longer than 128 characters. So yes, I used the web to search for images. However, I am still in search of a music service that would work well with Discord's limitations.

Well my point is that maybe the best solution is doing what you did as a proof of concept and using a localhost to send the image links to discord instead of finding such a service. That way you can use images already in local files as well as populating the host with images you find from the web without the link limitations. Unless I'm completely misunderstanding things of course, if so sorry I couldn't be of any help!

**Comment by iBowie** [comment]
> > > > > > Currently my main roadblock is that I can't find a cover art service which URL will not be longer than 128 characters.
> > > > > > I tried MusicBrainz with CovertArchive, but it requires THREE separate web requests and resulting link is too long for Discord to handle. Maybe someone could lend me a hand with this and help me find a service like this?
> > > > > > For the proof-of-concept I used my own VPS, just to make sure it will work.
> > > > > 
> > > > > 
> > > > > Does the largeimagekey accept any link? If the character size itself is the issue couldn't you use a link shortener? That would be even more web requests I guess but it may solve the problem. Does it only accept links? Could you perhaps index the queries into a local database and parse it from there? Not really familiar with the limitations of the Discord rich presence stuff but I'm just throwing out ideas to see if I could help in some way, sorry if my suggestions are uninformed.
> > > > 
> > > > 
> > > > Unfortunately, Discord requires direct image link, otherwise it wouldn't work.
> > > 
> > > 
> > > Have you tried Spotify or Youtube Music as a source for album covers? Also personally I have all my music downloaded with the covers in its metadata, is your implementation parsing local files or only searching the web for them?
> > 
> > 
> > Spotify has closed API, so does YouTube, and their links are most certainly longer than 128 characters (due to CDN and a lot of parameters and long names). You can't just send image to Discord as cover, since you can only use either pre-loaded Asset on Discord Developer Portal, or direct image link that is no longer than 128 characters. So yes, I used the web to search for images. However, I am still in search of a music service that would work well with Discord's limitations.
> 
> Well my point is that maybe the best solution is doing what you did as a proof of concept and using a localhost to send the image links to discord instead of finding such a service. That way you can use images already in local files as well as populating the host with images you find from the web without the link limitations. Unless I'm completely misunderstanding things of course, if so sorry I couldn't be of any help!

In that case you would need to boot a web-server each time you start AIMP, and also use port-forwarding, which would be a security issue. Best case scenario would be hosting a web-server that would do required requests for you on some VPS hosting. Doesn't require much CPU time, just quite cumbersome to setup, and not very user-friendly.

Even better scenario would be finding some open API that gives you album cover when you provide them with artist and album name.

**Comment by iBowie** [comment]
Or, you know, you could always do this, but be limited by 297 album covers (+3 for play, pause and default logo)

![image](https://github.com/Exle/AIMP-Discord-Presence/assets/31174978/77b5fc98-b473-4476-b7ba-bfd0ee7c15b8)

Which is absurd

**Comment by Chris3461** [comment]
> Or, you know, you could always do this, but be limited by 297 album covers (+3 for play, pause and default logo)
> 
> ![image](https://private-user-images.githubusercontent.com/31174978/302116617-77b5fc98-b473-4476-b7ba-bfd0ee7c15b8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc5NTI5OTYsIm5iZiI6MTcwNzk1MjY5NiwicGF0aCI6Ii8zMTE3NDk3OC8zMDIxMTY2MTctNzdiNWZjOTgtYjQ3My00NDc2LWI3YmEtYmZkMGVlN2MxNWI4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE0VDIzMTgxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU1N2ExNzQ3M2FmZDVkMTJhOWRlZTlmMzNkYjYxMWRiMGM3MGQwZDA3NDQ1MTdhYTk0ZjkyY2JkZWZiYjMwYzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ekH3Er_wuyDEh1hawjpAhgi88VPeXySSbPDtYwbJ89A)
> 
> Which is absurd

I hope you make it, good luck

**Comment by prototype464** [comment]
> Currently my main roadblock is that I can't find a cover art service which URL will not be longer than 128 characters.
> 
> I tried MusicBrainz with CovertArchive, but it requires THREE separate web requests and resulting link is too long for Discord to handle. Maybe someone could lend me a hand with this and help me find a service like this?
> 
> For the proof-of-concept I used my own VPS, just to make sure it will work.

What about providing the current file's metadata? Paired with something like MP3Tag for changing cover art that would be a decent compromise, doesn't have to use an online service for the time being

**Comment by iBowie** [comment]
> What about providing the current file's metadata? 

You HAVE to provide Discord with either image URL or image asset ID. You CAN'T send just raw image data to Discord.

**Comment by Zeninari** [comment]
does anyone know how MusicBee does their album art sending? maybe that is a hint to getting this working. i assume it caches the image somewhere then sends the URL.

is there any hope iBowie of getting this working at all?

**Comment by Zeninari** [comment]
im still impressed by the plugin even if it never happens but there has to be a way to do it right? if musicbee was able to make it work, then it must be possible to make AIMP also have it somehow. 

its an intresting issue indeed . . .

i decided to look into discord bee the plugin for musicbee:

Album Covers:
"If the cover upload feature is enabled your album covers will be uploaded to an anonymous Imgur album. This album is unique to every instance of the plugin, so nothing is shared between users of the plugin. You don't need an Imgur account for this feature to work."

this is interesting to me as i never thought about that. the links from imgur is normally short i believe. have you tried doing anything similar as a concept?

if we did something like this i would actually prefer if the bot we choose uploads them to our own imgur album (req account) if we chose to do that. mainly because that would make me feel better since some of my music has my own art on it.

just trying to think about if there is any way to save this discord presence since its VERY good in my opinion. it shows the most stuff live from what i can see. i havent found another discord presence maker that is better than this, - the album art of course xP

**Comment by iBowie** [comment]
> im still impressed by the plugin even if it never happens but there has to be a way to do it right? if musicbee was able to make it work, then it must be possible to make AIMP also have it somehow.
> 
> its an intresting issue indeed . . .
> 
> i decided to look into discord bee the plugin for musicbee:
> 
> Album Covers: "If the cover upload feature is enabled your album covers will be uploaded to an anonymous Imgur album. This album is unique to every instance of the plugin, so nothing is shared between users of the plugin. You don't need an Imgur account for this feature to work."
> 
> this is interesting to me as i never thought about that. the links from imgur is normally short i believe. have you tried doing anything similar as a concept?
> 
> if we did something like this i would actually prefer if the bot we choose uploads them to our own imgur album (req account) if we chose to do that. mainly because that would make me feel better since some of my music has my own art on it.
> 
> just trying to think about if there is any way to save this discord presence since its VERY good in my opinion. it shows the most stuff live from what i can see. i havent found another discord presence maker that is better than this, - the album art of course xP

Sounds like an idea, will try

**Comment by iBowie** [comment]
Tried it - it works!
At least with limited testing that I did.
![image](https://github.com/Exle/AIMP-Discord-Presence/assets/31174978/fb0b1807-c338-4bbc-907f-f39c14d99b3b)

**Comment by iBowie** [comment]
It works, however, there are few caveats I have to fix before it's ready for use.
* It tries to upload multiple times, all because of not async nature of the plugin
* I tried to implement a persistent cache, however it crashed the player during save (player kept playing, UI froze)

**Comment by Zeninari** [comment]
> It works, however, there are few caveats I have to fix before it's ready for use.
> 
>     * It tries to upload multiple times, all because of not async nature of the plugin
> 
>     * I tried to implement a persistent cache, however it crashed the player during save (player kept playing, UI froze)

good to see that it did work at all, if you can't get it fully working that's alright. the fact that you attempted and tried, that's the important part.

i also never mentioned what i use on mobile though i have no idea how it works. its called Kizzy (though i think it uses android features to push the album art and other info to discord. however it does use a PC to push it to discord . . . shown by the red desktop icon that appears when using kizzy(if also not on PC that is))

ill also say again i do not mind if we need an imugur account as that is something i would want to keep the art i made SAFE. ik it probably would be either way, but thats just a preference of mine.

**Comment by Zeninari** [comment]
> It works, however, there are few caveats I have to fix before it's ready for use.

i tried to wait from a response from you but ill ask now, do you have any ETA of when it might be done or in a somewhat operable state? or did an issue arise? i don't like rushing people but its more sad when there's no response.

**Comment by nicowinnerballs** [comment]
> It works, however, there are few caveats I have to fix before it's ready for use.

take your time dude, i attempted to do something like this a while back and as a complete amateur it crushed me. youre doing some great work, cant wait for a public release

**Comment by Zeninari** [comment]
> > It works, however, there are few caveats I have to fix before it's ready for use.
> 
> take your time dude, i attempted to do something like this a while back and as a complete amateur it crushed me. youre doing some great work, cant wait for a public release

indeed couldnt agree more. just wish they would give at least some sort of update. mainly because i'm getting a bit paranoid that this was dropped again which would be sad but id like to know that. either way i see ibowie did heart your comment and thats good enough for me!

i wish you luck ibowie you are doing a godsend for all of us that love aimp ^^

**Comment by Zeninari** [comment]
hope its going well, a new update from the original author kind of added album art but i really do not like how it is implemented. if you get it working from the tags (as well as keeping the original functions intact, like the play/pause status and the how much is left instead of how long the plugin has been active) then i will be 100% using this version of the plugin no questions asked.

i am so glad it renames the plugins to NAME.dll_old instead of overwriting it. i do not recommend updating the plugin if it still works for you as it does for me.

**Comment by iBowie** [comment]
> hope its going well, a new update from the original author kind of added album art but i really do not like how it is implemented. if you get it working from the tags (as well as keeping the original functions intact, like the play/pause status and the how much is left instead of how long the plugin has been active) then i will be 100% using this version of the plugin no questions asked.
> 
> i am so glad it renames the plugins to NAME.dll_old instead of overwriting it. i do not recommend updating the plugin if it still works for you as it does for me.

I did have some progress, and I'm pleased to say it's working quite well. All this time I was ironing some of the lesser issues, and implementing a lot of features. Here's how it looks now.
![image](https://github.com/user-attachments/assets/0f30979b-e653-4654-98d1-f7b00011bf0f)

Hope to get it as soon as possible, with guides how to setup album covers (it requires some one-time setup, hope it's worth it)

**Reply by Zeninari** [reply]
can't wait to see the progress it looks great!

at first i thought this was abandoned since Exle closed it making me quite sad. seeing what i do in that picture is amazing to see. really my only complaint was being unable to read from the tags. if musicbee can figure it out then im sure someone else could for aimp.

but ill say it again for you @iBowie if you manage to get this working (or at least get the album arts to be way more consistant to what is actually playing oml) then you will be quite litterally one of the best people for this plugin and would have done us all an amazing service. i can't wait for the next update now, especailly since i thought it was dead.

seeing this alive makes me incredibly happy.

**Reply by Zeninari** [reply]
Responses Merged:
> with guides how to setup album covers (it requires some one-time setup, hope it's worth it)

also for this as long as its automatic after the one time setup, idm. ive gone through a lot of painful setups some that takes multiple hours to get working. im sure the guide will be fine @iBowie 

Hopefully everything is well @iBowie its been a while, please don't let this hurt your health or anything. id just like to know if its been canceled since i do still come here checking for updates.

i hope your day/night is going well when you see this, same for other people as well.

**Reply by iBowie** [reply]
It's not cancelled, no, I was just busy with my other projects, and this one isn't such a big priority. I will at least publish the source code and binaries by the end of this year though.

**Reply by Zeninari** [reply]
good to hear thats a relief. i don't mind if you are still working on it im just glad that its not canceled. i think im probably the most hyped for this to get finished, but no rush!

i hope your other projects are going well then as well :3

**Reply by iBowie** [reply]
sadly, i don't think i'll be able to publish it this year, as it has a few flaws which are quite critical, one of them being getting rate-limited by imgur due to fast song switching, so i will have to make use of some sort of persistent storage for urls, as i initially tried to do, but that was somehow causing crashes to aimp itself. just wanted to let you know that i'm not forgetting or abandoning this project.

**Reply by Zeninari** [reply]
> sadly, i don't think i'll be able to publish it this year, as it has a few flaws which are quite critical, one of them being getting rate-limited by imgur due to fast song switching, so i will have to make use of some sort of persistent storage for urls, as i initially tried to do, but that was somehow causing crashes to aimp itself. just wanted to let you know that i'm not forgetting or abandoning this project.

its fine and that is i think how kizzy on android actually works is it stores URLS in a local folder but im not exactly sure how it works, funny enough works the way i wish aimp did ^^' Kizzy is on github just know that it is an android app if you want to look into how it works ;-;

either way your doing your best ibowie and thats all any of us can ask for. thanks for doing what you can for making the presence better for all of us. you really are doing amazing work i mean that.

**Reply by thealtersky** [reply]
For the album art, one way you could approach this is by using your own Discord account to upload the album art to a channel.

You would need to enter your Discord account token and channel ID, and then the plugin can upload the album art and use the image URL for the RPC. However, keep in mind that this might violate Discord's Terms of Service.

**Reply by iBowie** [reply]
And still, it wouldn't work because image URLs generated this way are too long and expire.

**Reply by iBowie** [reply]
It's a bad implementation due to the reason you've already said - it breaks discord TOS. And raises a security concerns for the user. I personally would never share my auth token with a third party plugin.

**Reply by JMiracIe** [reply]
What about using catbox / litterbox instead of imgur? 
foobar2k discord rpc component does it and works pretty well.

**Reply by Zeninari** [reply]
> What about using catbox / litterbox instead of imgur? foobar2k discord rpc component does it and works pretty well.

could be intresting to try if the imgur route isn't working. i told them about imgur since the musicbee plugin for discord uses imgur, but its really up to iBowie for what they want to do here. again the main issue is normal image links are way too long, and imgur has short links.

hope your doing well iBowie, and no pressure to get it done. i say to every other project i know about, idm waiting for as long as it takes since i know things like this take time. plus as you have told me before you have other things going on rn, and this project isn't a focus. 

edit: fixed run on sentence

**Reply by iBowie** [reply]
sounds like a plan, I'll give it a try, perhaps even a system that will use other service if one fails? sounds exciting

**Reply by ayukovt** [reply]
yo its been a while, any updates?

**Reply by nin-ocho** [reply]
Bump to that

**Reply by iBowie** [reply]
I will release source code shortly, there will be some instructions, but sadly don't expect much support since the free time I have is quite scarce.

**Reply by iBowie** [reply]
So, there it is: https://github.com/iBowie/AIMP-Discord-Presence-2

It's in no way can fully replace the plugin in question as my plugin works with .NET (as it's my language I'm most familiar with, but should still work)

You can find some docs in docs folder of the repo. You can also notice that the commit history is almost non-existent, it's due to me using API keys in private repo, so i had to re-create the repo as to not leak any API keys.

**Reply by JMiracIe** [reply]
So, as for now, Imgur is not allowing anyone to create new API keys, is there any workaround for this issue?

**Reply by iBowie** [reply]
I think they just made the process extra hard.

https://apidocs.imgur.com/#intro

**Reply by ayukovt** [reply]
> So, there it is: https://github.com/iBowie/AIMP-Discord-Presence-2
> 
> It's in no way can fully replace the plugin in question as my plugin works with .NET (as it's my language I'm most familiar with, but should still work)
> 
> You can find some docs in docs folder of the repo. You can also notice that the commit history is almost non-existent, it's due to me using API keys in private repo, so i had to re-create the repo as to not leak any API keys.

how exactly do you install this? i ran the publish scripts and tried selecting the output zip file in aimp to install it and it says it has an invalid file format

**Reply by ayukovt** [reply]
just noticed theres a pre-release download in the github repo, tried that too and it doesnt wanna install either

**Reply by iBowie** [reply]
Fixed the script for publishing and changed the release file to a proper format. It's been a while since I last worked on this plugin, so kinda forgot the format :p

**Reply by ayukovt** [reply]
> I think they just made the process extra hard.
> 
> https://apidocs.imgur.com/#intro

from what I've read, they actually did just stop providing new API keys (unless i don't know smth but from my research there's no way to generate new API keys anymore) so imgur unless they change something again, it's not a possible hosting option.

**Reply by iBowie** [reply]
From what I've read you have to use Postman now and make a web request directly to obtain a key, which is genuinely the worst thing they could come up with, if I'm right on this one.

**Reply by ayukovt** [reply]
> From what I've read you have to use Postman now and make a web request directly to obtain a key, which is genuinely the worst thing they could come up with, if I'm right on this one.

you still need the client ID and secret which you can only get by registering the app through imgurs website (which just redirects you to the main page now) so imgur isnt an option. Maybe you could add a selfhosted option tutorial or use catbox? I have my own website which I could utilize for personal use (or I could just use my local server for this) but I don't think I can do that without the plugin supporting it obv

**Reply by iBowie** [reply]
Yeah, I already made a self-hosting provider in the past, so should be fairly easy to bring it back (albeit with changes to support configuration, ofc).

**Reply by iBowie** [reply]
In the meantime, Discord App Assets could be considered as self-hosting, although the limit is 298 album covers.

**Reply by ayukovt** [reply]
> In the meantime, Discord App Assets could be considered as self-hosting, although the limit is 298 album covers.

I'm trying to get album covers to work through discord app assets and it seems like its failing to fetch it despite me doing everything exactly like the tutorial said. I can't really see plugin logs I think so I don't know where exactly its failing (its just displaying the rpc app icon i set instead of the album cover)

**Reply by Zeninari** [reply]
for some reason i cant get it to install it keeps saying its an invalid format, both with the entire main repo, and with ziping just the plugin part

Edit: i was able to get it to do something but its just giving an error that means the package is currupt: "Package is malformed (2)"

at this point i'm just going to give up trying to install this, either i'm too dumb or something else is going on i don't know all i know is that i'm frustrated

edit2: yeah i'm giving up, i can't deal with a plugin like this that wont install no matter what i do its either invalid or something else i can't get aimp to recognize this archive and its frustrating

**Reply by iBowie** [reply]
> for some reason i cant get it to install it keeps saying its an invalid format, both with the entire main repo, and with ziping just the plugin part
> 
> Edit: i was able to get it to do something but its just giving an error that means the package is currupt: "Package is malformed (2)"
> 
> at this point i'm just going to give up trying to install this, either i'm too dumb or something else is going on i don't know all i know is that i'm frustrated
> 
> edit2: yeah i'm giving up, i can't deal with a plugin like this that wont install no matter what i do its either invalid or something else i can't get aimp to recognize this archive and its frustrating

You're supposed to install it from the releases tab, not code. Also, it's only for 32 bit AIMP.

**Reply by iBowie** [reply]
At this point I will just give up with this version and go learn C++...

**Reply by iBowie** [reply]
> for some reason i cant get it to install it keeps saying its an invalid format, both with the entire main repo, and with ziping just the plugin part
> 
> Edit: i was able to get it to do something but its just giving an error that means the package is currupt: "Package is malformed (2)"
> 
> at this point i'm just going to give up trying to install this, either i'm too dumb or something else is going on i don't know all i know is that i'm frustrated
> 
> edit2: yeah i'm giving up, i can't deal with a plugin like this that wont install no matter what i do its either invalid or something else i can't get aimp to recognize this archive and its frustrating

I thought the issue arose from the latest build, but no, it's working for me as intended, it's also installing correctly. So even if this issue exists - it's from the fact that the plugin is only for 32bit AIMP.

The more glaring issue i found is that AIMP freezes on track change (which i just fixed, not published yet)

**Reply by iBowie** [reply]
Also, from this moment on I'd suggest using issues on the actual repository than continuing it here.

https://github.com/iBowie/AIMP-Discord-Presence-2/issues

**Reply by ayukovt** [reply]
> Also, from this moment on I'd suggest using issues on the actual repository than continuing it here.
> 
> https://github.com/iBowie/AIMP-Discord-Presence-2/issues

can i add you on discord? im trying to debug my issue with the plugin not fetching album covers and I don't really wanna create a bunch of new issues in your repo. I'm not like super skilled in C# but I am somewhat familiar with it so I thought i'd give it a shot

**Reply by naturbrilian** [reply]
> You're supposed to install it from the releases tab, not code. Also, it's only for 32 bit AIMP.

Sadly im using x64 version of Apps :<

**Reply by Zeninari** [reply]
> > for some reason i cant get it to install it keeps saying its an invalid format, both with the entire main repo, and with ziping just the plugin part
> > Edit: i was able to get it to do something but its just giving an error that means the package is currupt: "Package is malformed (2)"
> > at this point i'm just going to give up trying to install this, either i'm too dumb or something else is going on i don't know all i know is that i'm frustrated
> > edit2: yeah i'm giving up, i can't deal with a plugin like this that wont install no matter what i do its either invalid or something else i can't get aimp to recognize this archive and its frustrating
> 
> You're supposed to install it from the releases tab, not code. Also, it's only for 32 bit AIMP.

tbh i was tired and i wasnt in the best mood sorry ibowie, but yes i was being a bit dumb.

and i do use the 32 bit version because most plugins are made for the 32 bit version. ill look into installing it correctly this time ^^'

and no i was getting an error because i WAS trying to install it improperly like a dumb dumb, goes to show how bad the human mind can be when tired, i was able to fully install it properly this time with no issues by going to the releases and not by trying to bruteforce the code to function as a plugin (why i thought that would work? no idea)

**Reply by Zeninari** [reply]
> From what I've read you have to use Postman now and make a web request directly to obtain a key, which is genuinely the worst thing they could come up with, if I'm right on this one.

im not sure how it works but i was able to bypass postman entirely, plus the docs REQUIRES you to visit the main imgur website to make an api key and the link doesnt work. 

BUT!

if like me and you already have an application through imgur the client id still works as inteneded and i was able to get it working with my own Bot ID, and own Imgur ID:

<img width="438" height="160" alt="Screenshot 2025-12-09 002032" src="https://github.com/user-attachments/assets/6a0a23ae-8566-4e99-b17e-360c780e7389" />

again @iBowie i'm sorry for being stupid, i was tired and i apologize for the very harsh words ;-;

edit: [here's the direct link to check if you have an existing application](https://imgur.com/account/settings/apps)
again for some reason the [application creation link](https://api.imgur.com/oauth2/addclient) to add a application (which is required for postman) doesn't function and has been broken for a while. Unfortunately, i don't think there's currently a workaround currently. so unless you set it up before hand, you cannot use Imgur as a Client :(

**Reply by iBowie** [reply]
> > From what I've read you have to use Postman now and make a web request directly to obtain a key, which is genuinely the worst thing they could come up with, if I'm right on this one.
> 
> im not sure how it works but i was able to bypass postman entirely, plus the docs REQUIRES you to visit the main imgur website to make an api key and the link doesnt work.
> 
> BUT!
> 
> if like me and you already have an application through imgur the client id still works as inteneded and i was able to get it working with my own Bot ID, and own Imgur ID:
> <img alt="Screenshot 2025-12-09 002032" width="438" height="160" src="https://private-user-images.githubusercontent.com/165778697/524118023-6a0a23ae-8566-4e99-b17e-360c780e7389.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjUyODM0MDAsIm5iZiI6MTc2NTI4MzEwMCwicGF0aCI6Ii8xNjU3Nzg2OTcvNTI0MTE4MDIzLTZhMGEyM2FlLTg1NjYtNGU5OS1iMTdlLTM2MGM3ODBlNzM4OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIwOVQxMjI1MDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yMjZmNjkyN2NlMzZiMzc0YTJjN2M3ZjE3ZGU5ZDA2MGU4ZGYwZmY1NjhmMTM1NDVkMjk5NmMwMTM0NWQwMDA0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.zJh4zJ--EfgRclC_BH2f9xZOVDXX8lbp8H63QYN7HcQ">
> 
> again @iBowie i'm sorry for being stupid, i was tired and i apologize for the very harsh words ;-;
> 
> edit: [here's the direct link to check if you have an existing application](https://imgur.com/account/settings/apps) again for some reason the [application creation link](https://api.imgur.com/oauth2/addclient) to add a application (which is required for postman) doesn't function and has been broken for a while. Unfortunately, i don't think there's currently a workaround currently. so unless you set it up before hand, you cannot use Imgur as a Client :(

Glad you could figure things out! If there is anything confusing about the install, I could maybe adjust the installation guide accordingly.

**Reply by Zeninari** [reply]
> Glad you could figure things out! If there is anything confusing about the install, I could maybe adjust the installation guide accordingly.

again i am sorry for what i said i do mean that please ignore my incolence and stupidity from when i was tired.

the only thing i would say is to update and say the imgur method doesn't work for now unless you have an application already set up as the link to register an application via imgur isnt avaliable.

id also recommend pushing people to replace the BOT ID, so there's no rate limits (if everyone uses the same bot it can only update so fast, its why i use my own personal bots for this purpose)

other than that there was no problems with the install it was quite easy once i did what i was supposed to do lmao. feel free to link the page i linked before, again to check if you have an imgur application (idk if this requires something for the category to exist but):

https://imgur.com/account/settings/apps

if theres no applications under there you cant use imgur until whatever problem imgur is having regarding creating a new imgur app is fixed. as for postman thats only if you want to build something around an already existing imgur application thats been registered, its like a coding tool.

**Reply by Zeninari** [reply]
i frogot to also ask, does the imgur option automatically make sure its cleared every once in a while?

if its not then problems can arise (i also don't see a manual flush option like musicbee has though i probably can use musicbee to do so since ... it uses the same application lol)

either way this is just a simple ask of how exactly images are handled via imgur

**Reply by iBowie** [reply]
> i frogot to also ask, does the imgur option automatically make sure its cleared every once in a while?

Yes.
